cmd/goplugin-heroku/resource/app.go

```diff
@@ -0,0 +1,96 @@
+package resource
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-hclog"
+	heroku "github.com/heroku/heroku-go/v5"
+	"github.com/lyraproj/servicesdk/serviceapi"
+)
+
+type App struct {
+	ID           string
+	Locked       *bool
+	Maintenance  bool
+	Name         *string
+	Organization *string
+	Personal     bool
+	Region       *string
+	Space        string
+	Stack        *string
+}
+
+type AppHandler struct{}
+
+func (*AppHandler) Create(desiredState *App, service *heroku.Service) (*App, string, error) {
+	hclog.Default().Debug("Creating Heroku App", "desiredState", desiredState)
+
+	// Creates a resource and allocates an ID that can be used to read it in the future
+	app, err := service.OrganizationAppCreate(context.TODO(), heroku.OrganizationAppCreateOpts{
+		Locked:       desiredState.Locked,
+		Name:         desiredState.Name,
+		Organization: desiredState.Organization,
+		Personal:     &desiredState.Personal,
+		Region:       desiredState.Region,
+		Space:        &desiredState.Space,
+		Stack:        desiredState.Stack,
+	})
+	if err != nil {
+		hclog.Default().Error("Error creating app:", err)
+	}
+
+	desiredState.ID = app.ID
+
+	return desiredState, app.ID, nil
+}
+
+func (*AppHandler) Read(appID string, service *heroku.Service) (*App, error) {
+	hclog.Default().Debug("Reading App", "appID", appID)
+
+	app, err := service.OrganizationAppInfo(context.TODO(), appID)
+	if err != nil {
+		return nil, serviceapi.NotFound("App", appID)
+	}
+
+	actualState := App{
+		ID:           app.ID,
+		Locked:       &app.Locked,
+		Name:         &app.Name,
+		Organization: &app.Organization.Name,
+		Region:       &app.Region.Name,
+		Space:        app.Space.Name,
+		Stack:        &app.Stack.Name,
+	}
+
+	return &actualState, nil
+}
+
+func (*AppHandler) Update(appID string, desiredState *App, service *heroku.Service) (*App, error) {
+	hclog.Default().Debug("Updating App", "appID", appID, "desiredState", desiredState)
+
+	actualState, err := service.AppUpdate(context.TODO(), appID, heroku.AppUpdateOpts{
+		BuildStack:  desiredState.Stack,
+		Maintenance: &desiredState.Maintenance,
+		Name:        desiredState.Name,
+	})
+	if err != nil {
+		return nil, serviceapi.NotFound("App", appID)
```

Comment:
Looks like there's a risk of cloaking errors that are not due to `NotFound` here.

Reply:
Yeah, at the time I was just looking for a built-in error for now. Probably should have some real error handling in that.